pyproject.toml

```diff
@@ -47,7 +47,7 @@ dependencies = [
   'sqlalchemy~=2.0',
   'tabulate>=0.8.0,<0.10.0',
   'tqdm~=4.45',
-  'typing-extensions~=4.0;python_version<"3.10"',
+  'typing-extensions~=4.1;python_version<"3.11"',
```

Comment:
4.1 is the first version that had `assert_never` https://github.com/python/typing_extensions/blob/main/CHANGELOG.md#release-410-february-12-2022